src/utils/dimensionality_operations.jl

```diff
@@ -122,17 +126,22 @@ end
             column          patch       checkerboard
 
  *Output*:
-
+ If 4D tensor:
  - `X`: Reshaped tensor of dimensions `nx*2` x `ny*2` x `n_channel/4` x `batchsize`
+ If 5D tensor:
+ - `X`: Reshaped tensor of dimensions `nx*2` x `ny*2` x `nz*2` x `n_channel/8` x `batchsize`
 
  See also: [`squeeze`](@ref), [`wavelet_squeeze`](@ref), [`wavelet_unsqueeze`](@ref)
 """
 function unsqueeze(Y::AbstractArray{T,N}; pattern="column") where {T, N}
 
     # Dimensions
     batchsize = size(Y, N)
-    if any([mod(nn, 2) == 1 for nn=size(Y)[1:N-2]])
-        throw("Input dimensions must be multiple of 2")
+    if N == 4 && mod(size(Y)[N-1], 4) != 0
+        throw("With 4D tensor: number of channels must be divisible by 4")
+    end
+    if N == 5 && mod(size(Y)[N-1], 8) != 0
```

Comment:
`mod(size(Y, N-1), 2^(N-2)) != 0` and single check

Reply:
I knew I was missing something like that, thank you!